apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/converter/TimeDuration.java

```diff
@@ -48,23 +48,25 @@ public static TimeDuration of(String durationString) {
 
     private static int getDurationMultiplier(String unit) {
         switch (unit) {
-            case "ms":
+            case "us":
                 return 1;
-            case "s":
+            case "ms":
                 return 1000;
+            case "s":
+                return 1000 *1000;
             case "m":
-                return 1000 * 60;
+                return 1000 * 1000 * 60;
             default:
                 throw new IllegalStateException("Duration unit '" + unit + "' is unknown");
         }
     }
 
     public long getMillis() {
-        return durationMs;
+        return durationMicros/1000;
```

Comment:
`TimeDuration.of("999us").getMillis()` will be `0`. I don't think this is the expected behavior.

Reply:
I also stumbled upon this problem when implementing #2491. Unfortunately, I didn't find a good solution so far.

Reply:
personally I think that's correct in this situation as these are for configurations rather than statistics

Reply:
We're aiming to treat these differently with https://github.com/elastic/apm-agent-java/pull/2497 so you can only use microseconds if those are explicitly supported by the config - which for now will only be `exit_span_min_duration`

Reply:
Ok, understood. I will adapt my PR once https://github.com/elastic/apm-agent-java/pull/2497 is merged.